packages/flutter_tools/test/general.shard/asset_bundle_test.dart

```diff
@@ -152,6 +152,35 @@ flutter:
       ProcessManager: () => FakeProcessManager.any(),
     });
 
+    testUsingContext('throws ToolExit when directory entry contains invalid characters', () async {
+      globals.fs.file('.packages').createSync();
+      globals.fs.file('pubspec.yaml')
+        ..createSync()
+        ..writeAsStringSync(r'''
+name: example
+flutter:
+  assets:
+    - https://mywebsite.com/images/
+''');
+      final AssetBundle bundle = AssetBundleFactory.instance.createBundle();
+      expect(() => bundle.build(packagesPath: '.packages'), throwsToolExit(
+        message: 'Unable to search for asset files in directory path "https://mywebsite.com/images/". '
+        'Please ensure that this is valid URI that points to a directory that is '
+        'available on the local file system.\n'
+        'Error details:\n'
+        'Unsupported operation: Illegal character in path: https:',
+      ));
+    }, overrides: <Type, Generator>{
+      FileSystem: () {
+        final FileSystem result = MemoryFileSystem(
+          style: FileSystemStyle.windows,
+        );
+        result.currentDirectory = result.systemTempDirectory.createTempSync('flutter_asset_bundle_test.');
+        return result;
+      },
```

Comment:
Is it possible to mark a test to run only on Windows hosts?

If not, should I create a separate testing group for Windows in this file for the sake of readability?

Reply:
Actually, this test still won't work on non-Windows hosts because the file system is not the thing throwing the error.

Reply:
Addressed in [try fix test](https://github.com/flutter/flutter/pull/141214/commits/eaf0536ca817754f7a342835f39c153a2e7ac8d0)